features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/startup/RumAppStartupDetectorImpl.kt

```diff
@@ -0,0 +1,116 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.startup
+
+import android.app.Activity
+import android.app.ActivityManager
+import android.app.Application
+import android.os.Build
+import android.os.Bundle
+import com.datadog.android.core.internal.system.BuildSdkVersionProvider
+import kotlin.time.Duration.Companion.nanoseconds
+import kotlin.time.Duration.Companion.seconds
+
+private val START_GAP_THRESHOLD = 5.seconds
+
+internal class RumAppStartupDetectorImpl(
+    private val application: Application,
+    private val buildSdkVersionProvider: BuildSdkVersionProvider,
+    private val appStartupTimeProvider: () -> Long,
+    private val processImportanceProvider: () -> Int,
+    private val timeProviderNanos: () -> Long,
+    private val listener: RumAppStartupDetector.Listener
+) : RumAppStartupDetector, Application.ActivityLifecycleCallbacks {
+
+    private var numberOfActivities: Int = 0
+    private var isChangingConfigurations: Boolean = false
+    private var isFirstActivityForProcess = true
```

Comment:
```suggestion
    private var isChangingConfigurations: Boolean = false
    private var isFirstActivityForProcess: Boolean = true
```

just for the consistency: either we show type for both, or we don't